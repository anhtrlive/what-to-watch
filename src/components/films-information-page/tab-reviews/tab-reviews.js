import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

const Review = (props) => {
  let {comments} = props;

  return <div className="movie-card__reviews movie-card__row">
    <div className="movie-card__reviews-col">

      {comments.map((element) => {
        let option = {year: `numeric`, month: `long`, day: `numeric`};
        let {rating, comment, date, user} = element;
        let formatDate = new Date(date).toLocaleDateString(`en-EN`, option);
        return <div className="review" key={date + user.id}>
          <blockquote className="review__quote">
            <p className="review__text">{comment}</p>
            <footer className="review__details">
              <cite className="review__author">{user.name}</cite>
              <time className="review__date" dateTime="2016-12-24">{formatDate}</time>
            </footer>
          </blockquote>
          <div className="review__rating">{rating}</div>
        </div>;
      })}
    </div>
  </div>;
};

Review.propTypes = {
  comments: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    comments: state.comments,
  });
};

export {Review};
export default connect(mapStateToProps, null)(Review);
