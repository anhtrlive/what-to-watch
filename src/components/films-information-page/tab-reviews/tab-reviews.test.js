import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import {Review} from './tab-reviews';

Enzyme.configure({adapter: new Adapter()});

const comments = [{
  'id': 1,
  'user': {
    'id': 19,
    'name': `Christina`
  },
  'rating': 3.9,
  'comment': `It was well acted, directed.`,
  'date': `2020-08-05T20:09:08.102Z`
}, {
  'id': 2,
  'user': {
    'id': 14,
    'name': `Corey`
  },
  'rating': 7,
  'comment': `I really find it difficult to believe this movie`,
  'date': `2020-08-08T20:09:08.102Z`
}, {
  'id': 3,
  'user': {
    'id': 14,
    'name': `Corey`
  },
  'rating': 7.5,
  'comment': `Poised to be an instant classic`,
  'date': `2020-08-06T20:09:08.102Z`
}];

it(`button plus list`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Review
          comments={comments}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
