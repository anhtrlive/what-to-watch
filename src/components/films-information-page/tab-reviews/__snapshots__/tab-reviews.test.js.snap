// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button plus list 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          It was well acted, directed.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Christina
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            August 5, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.9
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          I really find it difficult to believe this movie
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Corey
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            August 8, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Poised to be an instant classic
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Corey
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            August 6, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.5
      </div>
    </div>
  </div>
</div>
`;
